js/save.js

```diff
@@ -67,19 +87,42 @@ function importFlowGram(file) {
     }
 }
 
-function openFromFile(file) {
+function importFunction(file) {
+    try {
+        const parsedFile = JSON.retrocycle(JSON.parse(file))
+        if (parsedFile.signingKey !== signingKey) {
+            swal("Flowgramming doesn't support that file!")
+            return
+        }
+        Object.keys(parsedFile.contexts).forEach((functionName) => {
+            addFunctionToTable(functionName)
+        })
+        parentWindowContexts = {
+            ...parentWindowContexts,
+            ...parsedFile.contexts,
+        }
+        window.opener.addNewContext(parsedFile.contexts)
+    } catch (e) {
+        swal('Something went wrong!')
+    } finally {
+        hideLoader()
+    }
+}
+
+function openFromFile(file, fn = false) {
     showLoader('Importing', 'Please wait while your flowgram is imported')
     setTimeout(() => {
-        importFlowGram(file)
+        if (fn) importFunction(file)
+        else importFlowGram(file)
     }, 1000)
 }
 
-function saveFile(filename, data) {
+function saveFile(filename, data, saveBtn = 'save-btn') {
     const file = new Blob([data], { type: 'application/json;charset=utf-8' })
     if (window.navigator.msSaveOrOpenBlob) {
         window.navigator.msSaveOrOpenBlob(file, filename)
     } else {
-        const saveButton = document.getElementById('save-btn')
+        const saveButton = document.getElementById(saveBtn)
         saveButton.href = URL.createObjectURL(file)
         saveButton.download = filename
```

Comment:
The space between the delete button and the save button should be a little wider.